make/tests

```diff
@@ -106,7 +106,7 @@ else
 endif
 
 %.hpp-test : %.hpp test/dummy.cpp
-	$(COMPILE.cpp) $(CXXFLAGS) -O0 -include $^ -o $(DEV_NULL) -Wunused-local-typedefs
+	$(COMPILE.cpp) $(CXXFLAGS) -O0 -include $^ -S -o $(DEV_NULL) -Wunused-local-typedefs
```

Comment:
I'm not sure where this rule is used, is there a reason for this change in this PR?

Reply:
It's a MinGW UCRT-specific bug, where the toolchain doesn't recognise the `nul` device for some reason: https://github.com/msys2/MINGW-packages/issues/10730

Otherwise it fails with:
```
Assembler messages:
Fatal error: can't create nul: Invalid argument
```

But I should have included it only for Win+UCRT, so I've fixed that now

Reply:
I can't replicate the failure locally, which makes me think it's related to either:

 * rtools43 uses gcc 13.2 while GHA uses 12.2
 * rtools43 gcc built by MSYS2, GHA by Mingw-W64
 
 RTools43 g++:
```
g++.exe (Rev2, Built by MSYS2 project) 13.2.0
```

GHA g++:
```
g++.exe (x86_64-posix-seh-rev2, Built by MinGW-W64 project) 12.2.0
```